### PR TITLE
Update go-containerregistry

### PR DESCRIPTION
**Pull request by dgageot**
I don't know what I'm doing here. 

@jonjohnsonjr could you please take a look at what I had to change in pkg/skaffold/docker/image.go?
Thanks!

Signed-off-by: David Gageot <david@gageot.net>

### CODE REVIEW

File: pkg/skaffold/docker/image.go [hunk 1]
```diff
@@ -133,10 +132,7 @@ func addTag(ref name.Reference, targetRef name.Reference, auth authn.Authenticat
 		return err
 	}
 
-	wo := remote.WriteOptions{
-		MountPaths: []name.Repository{ref.Context()},
-	}
-	return remote.Write(targetRef, img, auth, t, wo)
+	return remote.Write(targetRef, img, auth, t, remote.WriteOptions{})
```
Comment (dlorenc) [inline pkg/skaffold/docker/image.go]: I think we want to keep MountOptions if we can.

Comment (dgageot) [reply pkg/skaffold/docker/image.go]: MountPaths field was removed. I don't know if something else needs to be done. @jonjohnsonjr @mattmoor ?

Comment (jonjohnsonjr) [reply pkg/skaffold/docker/image.go]: The happy path for mounting is now handled by the library, see https://github.com/google/go-containerregistry/pull/186

tl;dr: remote layer reads retain the source repo and we specialize that case on remote layer writes.

**Comment by jonjohnsonjr** [comment]
LGTM

File: pkg/skaffold/docker/image.go (hunk 1 above)
Comment (dlorenc) [reply pkg/skaffold/docker/image.go]: Ah, nice. Missed that change in the library somehow.
